python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -48,9 +70,181 @@ def from_model(
+            ...         [
+            ...             tf.keras.layers.InputLayer(input_shape=()),
+            ...             tf.keras.layers.Flatten(),
+            ...             tf.keras.layers.Dense(10),
+            ...             tf.keras.layers.Dense(1),
+            ...         ])
+            ...     model.save("my_model")
+            ...     checkpoint = TensorflowCheckpoint.from_saved_model("my_model")
+            ...     session.report({"my_metric": 1}, checkpoint=checkpoint)
+
+            >>> trainer = TensorflowTrainer(
+            ...     train_loop_per_worker=train_fn,
+            ...     scaling_config=ScalingConfig(num_workers=2))
+
+            >>> result_checkpoint = trainer.fit().checkpoint
+
+            >>> batch_predictor = BatchPredictor.from_checkpoint(
+            ...     result_checkpoint, TensorflowPredictor)
+            >>> batch_predictor.predict(ray.data.range(3))
+        """
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
+        checkpoint = cls.from_directory(dir_path)
+        checkpoint._flavor = cls.Flavor.SAVED_MODEL
         return checkpoint
 
-    def get_model_weights(self) -> tf.keras.Model:
-        """Retrieve the model weights stored in this checkpoint."""
-        model_weights, _ = _load_checkpoint(self, "TensorflowTrainer")
-        return model_weights
+    def get_model(
+        self,
+        model_definition: Optional[
+            Union[Callable[[], tf.keras.Model], Type[tf.keras.Model]]
+        ] = None,
+    ) -> tf.keras.Model:
+        """Retrieve the model stored in this checkpoint.
+
+        Args:
+            model_definition: This arg is expected if and only if
+                the checkpoint's flavor is `MODEL_WEIGHTS`.
+
+        Returns:
+            A Tensorflow Keras model.
+        """
+        if self._flavor is self.Flavor.MODEL_WEIGHTS:
+            if not model_definition:
+                raise ValueError(
+                    "Expecting input of `model` argument when checkpoint's flavor "
+                    "is `MODEL_WEIGHTS`."
+                )
+            model_weights, _ = _load_checkpoint_dict(self, "TensorflowTrainer")
+            model = model_definition()
+            model.set_weights(model_weights)
+            return model
+        else:
+            if model_definition:
+                warnings.warn(
+                    "Ignoring `model argument` when checkpoint's"
+                    " flavor is not `MODEL_WEIGHTS`."
+                )
+            get_model = None
+            if self._flavor == self.Flavor.H5:
+
+                def get_model(path: str):
+                    # Walk the directory to find an .h5 path
+                    # There should be one and only one that is .h5.
+                    h5_path = None
+                    files = os.listdir(path)
+                    for file in files:
+                        if file.endswith(".h5"):
```

Comment:
should we just store the relative path as part of the checkpoint metadata during `from_h5` rather than finding it here?